src/core/BufferGeometry.js

```diff
@@ -765,40 +765,75 @@ THREE.BufferGeometry.prototype = {
 
 	},
 
-	merge: function ( geometry, offset ) {
+	merge: function( geometry ) {
 
 		if ( geometry instanceof THREE.BufferGeometry === false ) {
 
 			console.error( 'THREE.BufferGeometry.merge(): geometry not an instance of THREE.BufferGeometry.', geometry );
+
 			return;
 
 		}
 
-		if ( offset === undefined ) offset = 0;
-
 		var attributes = this.attributes;
 
+		if ( this.index ) {
+
+			var indices = geometry.index.array;
+
+			var offset = attributes[ 'position' ].count;
+
+			for ( var i = 0, il = indices.length; i < il; i ++ ) {
+
+				indices[ i ] = offset + indices[ i ];
+
+			}
+
+			this.index.array = Uint32ArrayConcat( this.index.array, indices );
+
+		}
+
 		for ( var key in attributes ) {
 
 			if ( geometry.attributes[ key ] === undefined ) continue;
 
-			var attribute1 = attributes[ key ];
-			var attributeArray1 = attribute1.array;
+			attributes[ key ].array = Float32ArrayConcat( attributes[ key ].array, geometry.attributes[ key ].array );
```

Comment:
Also (custom) `BufferAttribute` are not limited to `Float32Array`, see: https://github.com/mrdoob/three.js/blob/master/src/core/BufferAttribute.js#L307